internal/grpcutil/regex.go

```diff
@@ -20,9 +20,12 @@ package grpcutil
 
 import "regexp"
 
-// FullMatchWithRegex returns whether the full string matches the regex provided.
-func FullMatchWithRegex(re *regexp.Regexp, string string) bool {
+// FullMatchWithRegex returns whether the full text matches the regex provided.
+func FullMatchWithRegex(re *regexp.Regexp, text string) bool {
+	if len(text) == 0 {
+		return re.Match([]byte(text))
```

Comment:
Call `MatchString()` instead?

Reply:
And can you also add two tests (one fail one pass)? Thanks!

Reply:
Sure thing. Updated the API and added tests.

Reply:
Thanks!